content-cache/src/nginx_manager.py

```diff
@@ -299,6 +407,40 @@ def _get_upstream_config_keys(config: LocationConfig) -> tuple[nginx.Key, ...]:
     return tuple(keys)
 
 
+def _get_upstream_healthchecks_worker(upstream: str, config: LocationConfig) -> str:
+    """Create the lua script to perform the healthchecks on backends.
+
+    Args:
+        upstream: The upstream name.
+        config: The virtualhost config.
+
+    Returns:
+        A string with the lua script for the healthcheck workers.
+    """
+    return rf"""local hc = require "healthcheck"
+
+        local ok, err = hc.spawn_checker{{
+            shm = "healthcheck",
+            upstream = "{upstream}",
+            type = "{config.protocol.value}",
+
+            http_req = "GET {config.healthcheck_path} HTTP/1.0\r\nHost: {config.hostname}\r\n\r\n",
+
+            port = {433 if config.protocol.value == "https" else 80},
+            interval = {config.healthcheck_interval},
+            timeout = 1000,
+            fall = 3,
+            rise = 2,
+            valid_statuses = {{200}},
+            concurrency = 10,
+        }}
+        if not ok then
+            ngx.log(ngx.ERR, "failed to spawn health checker: ", err)
+            return
+        end
+    """
```

Comment:
Perhaps you want to use https://docs.python.org/3.6/library/textwrap.html#textwrap.dedent so that it is nicely indented when written to a file

Reply:
Good point.
As for the NginxLuaSection class: I wanted first to get feedback on the approach toward LUA code => is it the way we want to go.

Reply:
As the code is included in a section in nginx, it actually looks better with the current indentation. So I'd keep it like this.